move/interchain_token_service/sources/versioned/interchain_token_service_v0.move

```diff
@@ -200,6 +211,58 @@ module interchain_token_service::interchain_token_service_v0 {
         (token_id, treasury_cap_reclaimer)
     }
 
+    public(package) fun link_coin(
+        self: &InterchainTokenService_v0,
+        deployer: &Channel,
+        salt: Bytes32,
+        destination_chain: String,
+        destination_token_address: vector<u8>,
+        token_manager_type: u8,
+        link_params: vector<u8>,
+    ): MessageTicket {
+        assert!(destination_token_address.length() != 0, EEmptyTokenAddress);
+
+        // Custom token managers can't be deployed with native interchain token type, which is reserved for interchain tokens
```

Comment:
add a `from_u256` to TokenManagerType enum to validate that it's a correct value, and then check if it matches native interchain token type